code/modules/mining/machine_vending.dm

```diff
@@ -172,7 +209,7 @@
 	if(href_list["purchase"])
 		if(istype(inserted_id))
 			var/datum/data/mining_equipment/prize = locate(href_list["purchase"])
-			if(!prize || !(prize in prize_list) || prize.cost > inserted_id.mining_points)
+			if(!prize || prize.cost > inserted_id.mining_points)
```

Comment:
This is exploitable atm. Very minor exploit but.
In theory people can give a reference to a prize that is not part of this machine. For instance a grey slime extract from a golem vendor. If they somehow get that reference.